src/components/shared/StackSearchField.js

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+import IdleSearchField from './IdleSearchField';
+import { changeCallTreeSearchString } from '../../actions/profile-view';
+import {
+  getCurrentSearchString,
+  getSearchStrings,
+} from '../../reducers/url-state';
+
+import './StackSearchField.css';
+
+type Props = {|
```

Comment:
This is like Gandalf yelling "YOU SHALL NOT PASS (unless correctly typed)" 

<img width="470" alt="image" src="https://user-images.githubusercontent.com/1588648/33284356-73124f36-d374-11e7-8f4b-a2166d4b7a1f.png">